### PR TITLE
community[minor]: feat: Layerup Security integration

### PR DESCRIPTION
**Pull request by JamsheedMistri**
**Description**: adds integration with [Layerup Security](https://uselayerup.com). Docs can be found [here](https://docs.uselayerup.com). Integrates directly with our TypeScript SDK.

**Note**: all methods for our product require a paid API key, so I only included 1 test which checks for an invalid API key response. I have tested extensively locally.

**Twitter handle**: [@layerup_](https://twitter.com/layerup_)

### CODE REVIEW

File: libs/langchain-community/src/llms/layerup_security.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import { LLM, type BaseLLMParams } from "@langchain/core/language_models/llms";
```
[the remark this answers is not in the document]
Comment (JamsheedMistri) [reply libs/langchain-community/src/llms/layerup_security.ts:1]: Resolved this in [853f867](https://github.com/langchain-ai/langchainjs/pull/4929/commits/853f867511d8bf4f8eeed9cc072a45556c86b861)!

File: docs/core_docs/sidebars.js [hunk 2]
```diff
@@ -329,6 +329,21 @@ module.exports = {
           items: [{ type: "autogenerated", dirName: "integrations/stores" }],
           link: { type: "doc", id: "integrations/stores/index" },
         },
+        {
```
Comment (jacoblee93) [inline docs/core_docs/sidebars.js:332]: Remove as discussed

File: libs/langchain-community/package.json [hunk 3]
```diff
@@ -200,6 +200,7 @@
     "@google-ai/generativelanguage": "^0.2.1",
     "@gradientai/nodejs-sdk": "^1.2.0",
     "@huggingface/inference": "^2.6.4",
+    "@layerup/layerup-security": "*",
```
Comment (jacoblee93) [inline libs/langchain-community/package.json:203]: Add as a devdep too, and add under `peerDependenciesMeta`

**Comment by jacoblee93** [comment]
Thanks, see comments and don't forget to run `yarn format` + `yarn lint`

File: libs/langchain-community/langchain.config.js [hunk 4]
```diff
@@ -97,6 +97,7 @@ export const config = {
     "llms/watsonx_ai": "llms/watsonx_ai",
     "llms/writer": "llms/writer",
     "llms/yandex": "llms/yandex",
+    "llms/layerup_security": "llms/layerup_security",
```
Comment (jacoblee93) [inline libs/langchain-community/langchain.config.js:100]: You will probably want a chat model variant in the near future

Comment (jacoblee93) [reply libs/langchain-community/langchain.config.js:100]: All the good models are chat models

File: libs/langchain-community/src/llms/layerup_security.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+import {
+  GuardrailResponse,
+  LayerupSecurity as LayerupSecuritySDK,
+  LLMMessage,
+} from "@layerup/layerup-security";
+
+export interface LayerupSecurityOptions extends BaseLLMParams {
+  llm: LLM;
```
Comment (jacoblee93) [inline libs/langchain-community/src/llms/layerup_security.ts:9]: Should be `BaseLLM`

**Comment by jacoblee93** [comment]
Thanks for your patience! Small typing issue, I would fix but I can't make edits here.

**Comment by jacoblee93** [comment]
Seems like you need to run `yarn` again to update the lockfile

I am happy to polish this up if you give me push access to the branch

**Comment by JamsheedMistri** [comment]
> Seems like you need to run `yarn` again to update the lockfile
> 
> I am happy to polish this up if you give me push access to the branch

I tried this but it didn't do anything. Added you to the repo!

**Comment by jacoblee93** [comment]
I think there's something wrong with the SDK types:

```
Error: @langchain/community:build: ../../node_modules/@layerup/layerup-security/types.ts(1,44): error TS2307: Cannot find module 'openai/resources' or its corresponding type declarations.
```

**Comment by JamsheedMistri** [comment]
@jacoblee93 Looks like some parts of the repo had the `@layerup/layerup-security` SDK on version 1.5.6, which had a types import bug for OpenAI. I just bumped the remaining imports from 1.5.6 to 1.5.9 which should fix the issue.

**Comment by jacoblee93** [comment]
Still failing with:

```
Cannot find module 'openai/resources' or its corresponding type declarations.
```

You should also run `yarn` to update the lockfile if you update an SDK version

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**